hibernate-core/src/main/java/org/hibernate/type/spi/TypeConfiguration.java

```diff
@@ -209,6 +215,27 @@ public void sessionFactoryClosed(SessionFactory factory) {
 		// 		valid while the TypeConfiguration is scoped to SessionFactory
 	}
 
+	public String matchTypeContribution(Mapping mapping, int columnType) {
+		return basicTypeRegistrationContributions
+				.stream()
+				.filter( type -> type.getBasicType().getColumnSpan( mapping ) == 1 && type.getBasicType().sqlTypes( mapping ).length == 1 &&
+						type.getBasicType().sqlTypes( mapping )[0] == columnType )
+				.map( type -> type.getBasicType().getName() )
+				.findAny()
```

Comment:
Why are we caching BasicTypeRegistration when we only ever use its BasicType?  Also, we have access to all of this information already via the TypeConfiguration's BasicTypeRegistry.